### ISSUE TITLE
[Bug] Some lines in the line chart disappear progressively

### ISSUE THREAD

**Issue by jgoncab-arte** [apache/echarts#19958]
### Version

5.4.3

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=line-stack&version=5.4.3

### Steps to Reproduce

![image](https://github.com/apache/echarts/assets/169657500/cdedc5ad-3d76-47d8-bede-bc187e33f134)

Create a chart with more than one serie, a lot of data and this options:
```
      type: 'line',
      emphasis: {
        focus: 'self',
        blurScope: 'global'
      },
      showAllSymbol: true,
```
Then move the mouse over the lines to highlight them.

### Current Behavior

The lines begin to fade except when the mouse is over them. When the mouse is not over any lines, they remain blurry.
![image](https://github.com/apache/echarts/assets/169657500/2131ec8e-9ea3-485e-adbc-c508e01efb3e)

### Expected Behavior

The lines should come back when no line is highlighted.
![image](https://github.com/apache/echarts/assets/169657500/0ff71311-7c15-4460-8924-21916574db8a)

### Environment

```markdown
- OS: Window 10
- Browser: Chrome 125.0.6422.60
```

### Any additional comments?

The link does not always produce the error. The next html does:
[line-stack.txt](https://github.com/apache/echarts/files/15376053/line-stack.txt)
I don't know the reason. I think the only difference is data.

**Comment by helgasoft** [comment]
bug confirmed - entire series line segments are not restored after emphasis blur. Only their symbols remain.
Tried with ```blurScope: 'coordinateSystem'```, but result is the same. 
Another workaround to try is ```xAxis.type:'time'``` - that means rewriting all data as datasets.

**Comment by jgoncab-arte** [comment]
I tried it with dataset but the bug is still there:
[blur-error-using-dataset.txt](https://github.com/apache/echarts/files/15387268/blur-error-using-dataset.txt)

**Comment by helgasoft** [comment]
yes, dataset shows the same - [Demo](https://echarts.apache.org/examples/en/editor.html?code=PTAEHMFMDtIJwIYBdKiQSwLaoCbIQLABQAZgK7QDGGA9tKHigM4CS0ASgtFABRNII4SACLJIAGlAwcolAEpQAb2KhQlOvwb5QAXlABtALoBuFaHQlQPdEwByCWzwAOgppDZIe02ZDl-lZqokNHBWjKh6_IIiYsZaKKAAPHresfGQAHRuMSg84RlQOZA8CgDUoACM_spEqnXxCBlOZEwAFjz6sADuoD55YnKSALLIrRmI0Dg0mCWgAFSVAAyLhnKmtXUAvmabUgA2bgEbqnuQSGpkcHAwRbqgUUI-6_XBodZ3i3HoSSmTT-alUoKGr1VSMRrNNodQKg7q9MQ8SiXa7QIpyDJIGgsADKAHlsUg4OhuCUsnt0JRiotxBVFoMYfURkgxhMpjMFAtadSGXUmSyuGzZpzlqBylzxDzVHzxgLpkKlotRaAAEzLCXHUHS1lyjkKpUAZjVktAWtl7PmevKABYjRrGaMZZMdRauUqAKy20G8h3a83CxXlABsyx5q2eoKRVxuPiyZz6kZRRQKcYRZUqaxh2w2WdU1yQl3o4PWWZAeAAnuTNHpwqwOFxeHC-gByVXK5UAWkWFXbFQAnE3BqBGwiW4s2539T3-351uXK-dq2Ja5xuMVh7lR-Ou1OB5IAOwejNEYhuOBMO76DJXgCCVwQZZ4e_RAGtIGWmCVDBlMAgnDwrOgCg6AAfEcuZnAWYF1EgZZOJAABcoBNuSsBNpI0AINgiHIU2pToJIbQ0F02JlpgABGNB7IhhJkBIMKQJgTitAgTA2IhiigMESJMNhbh7CQaGgGReyXNi6hwdh4B7DQZEIHsTagJs6r1CAhFdNeex7CR5GUYhJByW4kggHclBiOAIRlqAdB7GWMIgOo0CwNQthkJpPE0XRGowOoOAIUooAAB7YQFgllrx8DoJATAAPoKeU3w5opxZHsQNBOLQ9B6CCoBcFgyDoHQekGRIoCnFAkzwRxfGQNQkA4JVyEVE28EebsSlmAF14BWx_kwRJSEYNggkwgg3VMAAMggZGQFRUH1AAWq8P5ICgcDYYoZabWWmztooQxDDtig4DgmzDXaoBwDQAgoPBVpum1ymgGWXU9Yo7UbCAmKURgTjsfcrREdRcC0ZIS3IKt8FNooZGnYpj3gtk7EwjgWAwKxGiIfoPIcRhWFISFkh9X5TaDZATbvV6TanpFMWLGhPJUxFUXRU1j31IzRLM8q9PnRzNPRfqPOU9TzNWkLoJ88zbri-zIsxYGMt1JLMV7k2MKGGz9w0JclKIXONhIDsj1y4hp5MMQmxAA)
Hovering series (entering/exiting from either side) will gradually remove all lines. Disabling symbols ```showSymbol:false``` may help a little.
Any refresh, like resize or legend click, will restore the lines.

My guess is blur line redrawing is **interrupted** when there are many line series. 
One workaround is to use _legend_ instead of _emphasis_ to show/hide line series. Of course it becomes inconvenient when many series are overlapping, like in your example.

**Comment by jgoncab-arte** [comment]
Yes, I could see the change with `showSymbol: false` but I wanted to use the tooltip and it doesn't work properly without the symbols (similar to this https://github.com/apache/echarts/issues/19608#issuecomment-1937838187) .

I tried with `select` but according to this issue it doesn't work (https://github.com/apache/echarts/issues/16545) and will never work.

Any other idea? Is there any other way to refresh the chart? I'm also using `datazoom` in my case, so I need the update to keep the selection.

**Comment by helgasoft** [comment]
> I tried with `select` but according to this issue it doesn't work (#16545) and will never work.

good that you mentioned it! See my Demo Code in related #19194. Event [selectchanged](https://echarts.apache.org/en/api.html#events.selectchanged) should successfully replace emphasis in your example. Just remove all _areaStyle_. Toggle _lineStyle_ by clicking on a line symbol and mouse-out. 🥳

**Comment by jgoncab-arte** [comment]
Doesn't using `setOption` constantly affect performance? I would like the other series to blur and maybe I can make it work with that approach, but `setOption` would refresh the `dataZoom` selection constantly. Can I avoid this somehow?

**Comment by jgoncab-arte** [comment]
I made a test with this code:
```
chart.dispatchAction({type: 'select', seriesIndex: seriesIndex, dataIndex: dataIndex});
var options = chart.getOption();
for (let i = options.series.length - 1; i >= 0; i--) {
    if (i === seriesIndex) {
        options.series[i].lineStyle = {
            opacity: 1
        };
    } else {
        options.series[i].lineStyle = {
            opacity: 0.3
        };
    }
}
chart.setOption(options);
```
The `dataZoom` does not change, but the performance is really bad in my example.

**Comment by helgasoft** [comment]
I believe the best workaround is with toggle _click_ event - [Demo](https://echarts.apache.org/examples/en/editor.html?code=PTAEHMFMDtIJwIYBdKiQSwLaoCbIQLABQAZgK7QDGGA9tKHigM4CS0ASgtFABRNII4SACLJIAGlAwcolAEpQAb2KhQlOvwb5QAXlABtALoBuFaHQlQPdEwByCWzwAOgppDZIe02ZDl-lZqokNHBWjKh6_IIiYsZaKKAAPHresfGQAHRuMSg84RlQOZA8CgDUoACM_spEqnXxCBlOZEwAFjz6sADuoD55YnKSALLIrRmI0Dg0mCWgAFSVAAyLhnKmtXUAvmabUgA2bgEbqnuQSGpkcHAwRbqgUUI-6_XBodZ3i3HoSSmTT-alUoKGr1VSMRrNNodQKg7q9MQ8SiXa7QIpyDJIGgsADKAHlsUg4OhuCUsnt0JRiotxBVFoMYfURkgxhMpjMFAtadSGXUmSyuGzZpzlqBylzxDzVHzxgLpkKlotRaAAEzLCXHUHS1lyjkKpUAZjVktAWtl7PmevKABYjRrGaMZZMdRauUqAKy20G8h3a83CxXlABsyx5q2eoKRVxuPiyZz6kZRRQKcYRZUqaxh2w2WdU1yQl3o4PWWbwAE9yZo9OFWBwuLw4X0AOSq5XKgC0iwqbYqAE5G4NQAB2D0ZojENxwJh3fQZWcAQSuCFLPEH6IA1pBS0wSoYMpgEE4eFZ0AodAA-I65s4Fy91JClpyQABcoEb5Ngjck0AQ2Bfb8bpToJIbQ0F02KlpgABGNB7C-hJkBIMIgJAmBOK0CBMDYL6KKAwRIkwf5uHsJCfqAkF7Jc2LqI-f7gHsNCQQgeyNqAmzqvUhLoOAUBwAAMsSkAAKIAG43HBcAIexdREZA1CQDgQw0Dgz6vhO6CQEwn4wjA6jKdhoAAB5_gZpGloR8DqUwAD6LHlN8OascWo7EDQTi0PQeggqAXBYMg6B0C-JBMW4khmKcUCTE-OEyXJOBRW-FSNk-8GQLsbFmAZc4GVhShoA-KmNhg2CkQg2VMLxCCQZAsG5XANACCgT5Wm6aVSaWWU5Yo6UbCAmIwRgTj6SBXTiZJuEhPuSAoHAT6NookGbCx3Vgvg2TYTCOBYDAmEaC--g8jh36_q-JmSPeNGvkVkCNstoKNmpGlWYsWl2qpFmPYlUn1Pd73WcqL1ej9RKPfqAN3Q91lWmD30Q1ZbrQ3UQOWVZgYI6oSOPYOjYwoYX33DQlyUi-ZYVkgOxSRDL4TkwUnggAWjQ0zYdIT4elmWbjsSdzQGQex7OsmClgAwuhQgZHQPCNpQ5KUGupHOKeF44WYwmCKArnuVOeiCyL0TJkguJuf50AlOGFhWJh0AAIQ6DzfPAnUIDXNkMIa8bTCxsDHuvIJCCUO0FtyOet71B776QAS5aQJ5rl--g94vhUmzhqHGTxyhkenDHLiUPHZmVMnmagKOofEnodv8zs-yHF5qggJbdxOLGpyxfoKye5ZbDKQZcQgD9LdIP7dbydjGoN3oTcQ13kA93XYDSxSa4wq8VinOc3x6G7GgdxpGSnNwzKgG2lRfKAZ56J85htm2wI8ub7w6I_9zErfr2qFv0AexD-joLu4eZxEXKsdc4J0qI5Hkuxqo12NB_L-v0f5_wEgAu4OFgF5xfIsDI-pwFv3Vkbbe39f5pxQJgZBnk8Fx1AZg7BhdXoOVUA5ByOtRZIFjAbfBJtYGjk2GsIAA).
Event _selectchanged_ works only with line symbols, but not line segments - maybe a valid **new Feature** request.
And since lines here are so dense, it's better to be able to click on both.
![image](https://github.com/apache/echarts/assets/13038071/01ec50d1-fc79-48e3-9019-219ed4c11d54)

**Comment by jgoncab-arte** [comment]
It's not exactly the behavior I was asked to develop. Do you intend to fix this bug at some point?

**Comment by helgasoft** [comment]
> Do you intend to fix this bug at some point?

Cannot speak for the developers, but suspect they have bigger fish to fry. Performance bugs could be hard or even impossible to fix. I stick to workarounds and move on.
